api/hypershift/v1beta1/hosted_controlplane.go

```diff
@@ -41,6 +41,7 @@ type HostedControlPlane struct {
 // HostedControlPlaneSpec defines the desired state of HostedControlPlane
 // +kubebuilder:validation:XValidation:rule="self.platform.type == 'IBMCloud' ? size(self.services) >= 3 : size(self.services) >= 4",message="spec.services in body should have at least 4 items or 3 for IBMCloud"
 // +kubebuilder:validation:XValidation:rule="!has(self.operatorConfiguration) || !has(self.operatorConfiguration.clusterNetworkOperator) || !has(self.operatorConfiguration.clusterNetworkOperator.disableMultiNetwork) || !self.operatorConfiguration.clusterNetworkOperator.disableMultiNetwork || self.networking.networkType == 'Other'",message="disableMultiNetwork can only be set to true when networkType is 'Other'"
+// +kubebuilder:validation:XValidation:rule="!has(self.networking) || !has(self.networking.allocateNodeCIDRs) || self.networking.allocateNodeCIDRs == 'Disabled' || self.networking.networkType == 'Other'",message="allocateNodeCIDRs can only be set to Enabled when networkType is 'Other'"
```

Comment:
Does it make sense to have `allocateNodeCIDRs` with another networkType? Do you think this is more accurate:

```
!has(self.networking) || !has(self.networking.allocateNodeCIDRs) || self.networking.allocateNodeCIDRs != 'Enabled' || self.networking.networkType == 'Other'
```

Yours is more permissive I want to make sure we don't have gaps in the API :)